Lib/test/test_codecs.py

```diff
@@ -3440,5 +3440,22 @@ def search_function(encoding):
         self.assertEqual(NOT_FOUND, codecs.lookup('a\xe9\u20ac-8'))
 
 
+class EncodingNormalizationTest(unittest.TestCase):
+
+    def test_bpo39337(self):
+        """
+        bpo-39337: similar to _Py_normalize_encoding(),
+        encodings.normalize_encoding() should ignore non-ASCII letters.
+        """
```

Comment:
Your test is not limited to non-ASCII cases. I suggest to rename the test to "test_normalization" and move the comment on the test on non-ASCII characters.

Don't mention the private function _Py_normalize_encoding(), just say that "encodings.normalize_encoding() ignores non-ASCII letters". "should" is vague, I prefer to avoid it.